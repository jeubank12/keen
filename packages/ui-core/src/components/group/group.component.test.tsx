import React from 'react';
import { render } from 'enzyme';
import 'jest-styled-components';

import Group from './group.component';

describe('@keen.io/charts - <Group />', () => {
  it('should wrap groupped children', () => {
    const wrapper = render(
      <Group>
        <div>1</div>
        <div>2</div>
        <div>3</div>
        <div>4</div>
      </Group>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should wrap children with provided node', () => {
    const CustomNode = ({ children }: { children: React.ReactNode }) => (
      <span>{children}</span>
    );

    const wrapper = render(
      <Group groupNode={CustomNode} chunks={3}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
        <div>4</div>
      </Group>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should wrap groupped children based on chunks property', () => {
    const wrapper = render(
      <Group chunks={3}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
        <div>4</div>
      </Group>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
