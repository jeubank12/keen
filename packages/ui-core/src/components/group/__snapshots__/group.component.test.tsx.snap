// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@keen.io/charts - <Group /> should wrap children with provided node 1`] = `
Array [
  <span>
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
  </span>,
  <span>
    <div>
      4
    </div>
  </span>,
]
`;

exports[`@keen.io/charts - <Group /> should wrap groupped children 1`] = `
Array [
  <div
    class=""
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>,
  <div
    class=""
  >
    <div>
      3
    </div>
    <div>
      4
    </div>
  </div>,
]
`;

exports[`@keen.io/charts - <Group /> should wrap groupped children based on chunks property 1`] = `
Array [
  <div
    class=""
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
  </div>,
  <div
    class=""
  >
    <div>
      4
    </div>
  </div>,
]
`;
